src/compute/src/compute_state.rs

```diff
@@ -624,6 +624,22 @@ impl<'a, A: Allocate + 'static> ActiveComputeState<'a, A> {
                 .allows_reporting(&new_frontier)
                 .then(|| new_frontier.clone());
 
+            // Collect the output frontier and check for progress.
+            //
+            // By default, the output frontier equals the write frontier (which is still stored in
+            // `new_frontier`). If the collection provides a compute frontier, we construct the
+            // output frontier by taking the meet of write and compute frontier, to avoid:
+            //  * reporting progress through times we have not yet written
+            //  * reporting progress through times we have not yet fully processed, for
+            //    collections that jump their write frontiers into the future
```

Comment:
Maybe add something like "e.g., REFRESH MVs".

Reply:
Note that this can also happen with non-REFRESH MVs, if a slower replica observes advancements of the target shard's `upper` made by a faster replica.

I think it's good to mention all the known special cases in a single place, instead of repeating them in multiple places where they can go out of sync. I've added a blurb to the `FrontiersResponse` doc string, which is also where the meaning of an output frontier is specified.